File: companies/Old/gfg.cpp
int LCSof3(string A, string B, string C, int n1, int n2, int n3)
{
        int dino[n1 + 1][n2 + 1][n3 + 1];
        for (int i = 0; i <= n1; i++)
        {
            for (int j = 0; j <= n2; j++)
            {
                for (int k = 0; k <= n3; k++)
                {
                    if (i == 0 || j == 0 || k == 0)
                        dino[i][j][k] = 0;
                    else
                    {
                        if (A[i - 1] == B[j - 1] && B[j - 1] == C[k - 1])
                            dino[i][j][k] = 1 + dino[i - 1][j - 1][k - 1];
                        else
                            dino[i][j][k] = max({dino[i - 1][j][k], dino[i][j - 1][k], dino[i][j][k - 1]});
                    }
                }
            }
        }
        return dino[n1][n2][n3];
}

File: companies/Old/gfg.py
def LCSof3(n1, n2, n3):

    dp = [[[-1 for n1 in range(100)] for n2 in range(100)] for n3 in range(100)]

    if n1 == -1 or n2 == -1 or n3 == -1:
        return 0

    if dp[n1][n2][n3] != -1:
        return dp[n1][n2][n3]

    if A[n1] == B[n2] and B[n2] == C[n3]:
        dp[n1][n2][n3] = 1 + LCSof3(n1 - 1, n2 - 1, n3 - 1)
        return dp[n1][n2][n3]

    else:
        dp[n1][n2][n3] = max(
            max(LCSof3(n1 - 1, n2, n3), LCSof3(n1, n2 - 1, n3)), LCSof3(n1, n2, n3 - 1)
        )

        return dp[n1][n2][n3]
